oauth2client/client.py

```diff
@@ -194,15 +194,16 @@ class Credentials(object):
     NON_SERIALIZED_MEMBERS = frozenset(['store'])
 
     def authorize(self, http):
-        """Take an httplib2.Http instance (or equivalent) and authorizes it.
+        """Creates an http object that provides credentials to requests.
 
-        Authorizes it for the set of credentials, usually by replacing
-        http.request() with a method that adds in the appropriate headers and
-        then delegates to the original Http.request() method.
+        Typically will use :mod:`transport` to wrap the http object to provide
+        credentials to requests and handle refreshing credentials as needed.
 
         Args:
-            http: httplib2.Http, an http object to be used to make the refresh
-                  request.
+            http: The http object to wrap.
```

Comment:
It's ambiguous for now and will be clarified in later PRs. Essentially, every transport will have it's own "http object" (`httplib2.Http`, `urllib3.PoolManager`, `requests.Session`) and this function will eventually accept any http object for the supported transports.